Newsletters: add 337 (2025-01-17)

harding opened the pull request:
- [x] Lede, releases/RCs, topic entries @harding 
- [x] Notable merges @Gustavojfe

LarryRuane commented on _posts/en/newsletters/2025-01-17-newsletter.md:
```diff
@@ -0,0 +1,122 @@
+  thinks a pool could launch as a client of a larger pool, so even the
+  first miner to join the new pool would receive lower variance than
+  solo mining.  The intermediate pool could then sell the ecash shares
+  it earns in order to finance whatever payout scheme it chooses to pay
```
```suggestion
  it earns to finance whatever payout scheme it chooses to pay
```

LarryRuane commented on _posts/en/newsletters/2025-01-17-newsletter.md:
```diff
@@ -0,0 +1,122 @@
+  first miner to join the new pool would receive lower variance than
+  solo mining.  The intermediate pool could then sell the ecash shares
+  it earns in order to finance whatever payout scheme it chooses to pay
+  its individual miners.  Once the intermediate pool acquired a
```
```suggestion
  its miners.  Once the intermediate pool acquired a
```

LarryRuane commented on _posts/en/newsletters/2025-01-17-newsletter.md:
```diff
@@ -0,0 +1,122 @@
+  templates that suit its miners.
+
+- **Offchain DLCs:** developer conduition [posted][conduition offchain]
+  to the DLC-dev mailing list about a contract protocol that allows a
```
```suggestion
  to the DLC-dev mailing list about a contract protocol that allows an
```

LarryRuane reviewed (commented):
Small things; I'll look again after more content has been added. I always try to take a look Thursday (Mountain time) at around 11 PM when I get home from one of my jobs.

jirijakes commented on _posts/en/newsletters/2025-01-17-newsletter.md:
```diff
@@ -0,0 +1,148 @@
+  summary][news304 ecashtides] of a Delving Bitcoin thread about paying
+  [pool miners][topic pooled mining] with ecash for each share they
+  submitted.  Previously, Matt Corallo [asked][corallo whyecash] why a
+  pool would implement the extra code and accounting to handle ecash
```
```suggestion
  pool would implement the extra code and accounting to handle tradable ecash
```

Adds a bit of context regarding Matt's question.

bitschmidty commented on _posts/en/newsletters/2025-01-17-newsletter.md:
```diff
@@ -0,0 +1,148 @@
+  silent payments] using [PSBTs][topic psbt]. If there are multiple independent
+  signers, a DLEQ proof is required to allow all signers to prove to co-signers
+  that their signature is valid and doesn’t risk losing funds, without revealing
+  their private key (see Newsletter [#335][news335 dleq]).
```
In addition to news335 could also reference pod327  https://bitcoinops.org/en/podcast/2024/11/05/#draft-bip-for-dleq-proofs where we had Andrew Toth on to explain

bitschmidty reviewed (commented):
Notable code segment looks good. Just on small suggestion

harding commented:
Made edits for all feedback to date (thanks @bitschmidty @LarryRuane @jirijakes !), added lede, added topic entries, and reviewed @Gustavojfe section (great as always!).  I haven't yet updated releases/RCs.  I'll do that and another round of edits later.

bitschmidty reviewed (commented):
ACK topics and lede

bitschmidty commented on _topics/en/discrete-log-equivalency.md:
```diff
@@ -47,6 +47,9 @@ optech_mentions:
     url: /en/newsletters/2025/01/03/#bips-1689
 
+  - title: "BIPs #1687 adds BIP375 to specify sending DLEQs for multi-signer silent payments using PSBT"
+    url: /en/newsletters/2025/01/16/#bips-1687
```
```suggestion
    url: /en/newsletters/2025/01/17/#bips-1687
```

jirijakes commented on _posts/en/newsletters/2025-01-17-newsletter.md:
```diff
@@ -0,0 +1,154 @@
+  forcing pools to keep large reserves.  For example, if a pool miner
+  controls 1% of hashrate and creates shares on a template with about
+  1,000 BTC in fees and 3 BTC in subsidy, they would be owed by their
+  pool about 10 BTC.  However, if the pool does mine that block and,
+  when they do mine a block, fees are back down to a fraction of the
```
```suggestion
  pool about 10 BTC.  However, if the pool does mine a block and,
  when they do mine it, fees are back down to a fraction of the
```

This sentence is a bit confusing. As I understand it, “that block” refers to the block with 1000 BTC fees. But then the fees would not be down if the pool mines the same block, right? Shouldn't the pool mine some other block for this sentence to make sense?

LarryRuane replied:
I found this confusing too; if I understand correctly, the problem occurs when fees are high for a significant amount of time (so that miners' shares are for high-reward templates), but then when the pool finds a block, fees are much lower. If that's correct, maybe:

> If, however, fees drop to a fraction of the block reward just before the pool finds a block, the pool might have only 3 BTC to split among its miners, forcing ...

jirijakes commented on _posts/en/newsletters/2025-01-17-newsletter.md:
```diff
@@ -0,0 +1,154 @@
+  splice occurs, the parent channel's short channel identifier (SCID), capacity,
+  and balance bounds are updated instead of creating a new channel.
+
+- [Rust Bitcoin #3792][] adds support for [BIP324][]’s [v2 P2P transport][topic
+  v2 P2P transport] to allow Bitcoin nodes to communicate with each other over
+  encrypted connections (see Newsletter [#306][news306 v2]). This is achieved by
+  adding a `V2NetworkMessage` method, which wraps a `NetworkMessage` method and
+  handles v2 encoding and decoding.
```
I apologize for late review but I think this is not accurate. The PR adds only ability to understand (encode and decode) V2 P2P messages. It does not do anything above it, like encryption.

```suggestion
- [Rust Bitcoin #3792][] adds ability to encode and decode [BIP324][]’s [v2 P2P
  transport][topic v2 P2P transport] messages (see Newsletter [#306][news306 v2]).
  This is achieved by adding a `V2NetworkMessage` struct, which wraps the original
  `NetworkMessage` enum and provides v2 encoding and decoding.
```

harding replied:
Thanks for catching!

LarryRuane commented on _posts/en/newsletters/2025-01-17-newsletter.md:
```diff
@@ -0,0 +1,153 @@
+  all of its miners, forcing it to pay from its reserves.  If that
+  happens too many times, the pool's reserves will be exhausted and it
+  will go out of business.  Pools address this in various ways,
+  including using various [proxies for actual fees][news304 fpps proxy].
```
```suggestion
  including using [proxies for actual fees][news304 fpps proxy].
```
Probably don't want to use "various" twice so close together in the same sentence; I don't think the second one is needed.

LarryRuane commented on _posts/en/newsletters/2025-01-17-newsletter.md:
```diff
@@ -0,0 +1,153 @@
+  to the DLC-dev mailing list about a contract protocol that allows an
+  offchain spend of the funding transaction signed by both parties to
+  create multiple [DLCs][topic dlc].  After the offchain DLC has settled
+  (e.g., all required oracle signatures have been obtained), a new
+  offchain spend can be signed by both parties to reallocate funds
```
```suggestion
  (e.g., all required oracle signatures have been obtained), both parties can sign a new
  offchain spend to reallocate funds
```
I'm not very much against passive voice (I know it's common in technical writing), but just before this, we have "been obtained."

LarryRuane commented on _posts/en/newsletters/2025-01-17-newsletter.md:
```diff
@@ -0,0 +1,153 @@
+  `NetworkMessage` enum and provides v2 encoding and decoding.
+
+- [BDK #1789][] updates the default transaction version from 1 to 2 to improve
+  wallet privacy since prior to this, BDK wallets were more identifiable due to
```
```suggestion
  wallet privacy since, before this, BDK wallets were more identifiable due to
```
Or maybe "since previously, BDK wallets..." (I can't decide which I prefer)

LarryRuane commented on _posts/en/newsletters/2025-01-17-newsletter.md:
```diff
@@ -0,0 +1,153 @@
+- [BIPs #1396][] updates [BIP78][]’s [payjoin][topic payjoin] specification to
+  align with [BIP174][]’s [PSBT][topic psbt] specification, resolving a previous
+  conflict. In BIP78, a receiver previously deleted UTXO data after completing
+  its inputs, even if the sender needed the data. With this update, UTXO data is
+  now retained.
```
```suggestion
  its inputs, even if the sender needed the data. With this update, UTXO data will
  be retained.
```

harding commented on _posts/en/newsletters/2025-01-17-newsletter.md:
```diff
@@ -0,0 +1,159 @@
+  allow the same pool of funds to be used for both offchain DLCs and
+  LN (see Newsletters [#174][news174 dlc-ln] and [#260][news260 dlc]).
+  A [reply][conduition offchain2] from conduition described his
+  proposal's major difference from previous proposals.
```
Note that our summary doesn't go into any detail about what the major difference is.  I tried for over half an hour to understand it, and maybe I kinda got a glimmer of it, but I don't feel confident enough to try to summarize it at this late moment.  We'll have to trust that anyone who wants to learn more will follow the link.

harding commented:
Updated releases/RCs, added a coda for the DLC topic, and made some (but not all) final edits.  Thanks @bitschmidty @jirijakes @LarryRuane !

azuchi commented on _posts/en/newsletters/2025-01-17-newsletter.md:
```diff
@@ -0,0 +1,160 @@
+{% include snippets/recap-ad.md when="2025-01-21 15:30" %}
+{% include references.md %}
+{% include linkers/issues.md v=2 issues="3792,1789,1687,1396,2936" %}
+[ldk 0.1]: https://github.com/lightningdevkit/rust-lightning/releases/tag/v0.1
```
```suggestion
[ldk v0.1]: https://github.com/lightningdevkit/rust-lightning/releases/tag/v0.1
```

bitschmidty commented:
Added a topic update to be in line with the changes suggested in https://github.com/bitcoinops/bitcoinops.github.io/pull/2129#discussion_r1919545330

Squashed. Thank you all! 🚀
